publisher/webhookPublisher.go

```diff
@@ -116,6 +125,9 @@ func (p *WebhookPublisher) makeHttpRequest(r *publishRequest) {
 		})
 	} else {
 		log.Printf("Final retry failed, giving up on %v", url)
+		if r.respChan != nil {
```

Comment:
Hmm, we should not designate an arbitrary string as the error value.

We should have a Response struct, with an error code in it and a []byte for the response.  Then the error field can be used for indicating success or failure.